### PR TITLE
Add a dashboard in a new metrics.rs module

### PR DESCRIPTION
**Pull request by drozdziak1**
# Overview
This change introduces a simple HTML table for quick reference about individual symbol status. The table is served using a new HTTP endpoint, under "/dashboard". 

# Screenshot
[click here](https://imgur.com/a/kRtTGUx)

# Changes to existing code
* The HTTP endpoint is spawned in `agent.rs` together with other async jobs. It reuses the logger and some of the channel tx's.
* Macro recursion limit is raised to 256 (the new type-checked HTML crate is macro-heavy)
* New dependencies
* * typed-html - ergonomic HTML manipulation

# New code
* `metrics.rs` - a new metrics- and dashboard oriented module. I plan to also add Prometheus metrics there and I'll probably exile dashboard generation logic to something like `dashboard.rs`. lmk if it's not okay with you.

# Testing
* The existing Pytest harness continues to pass.
* I verified my code using the example publisher and the `integration-tests` config (that's also how I generated the screenshot). The timestamps and price values update often, as expected. The rightmost "last local update time" column displays "no data" unil updates start flowing in from example publisher, as expected. In the screenshot you can notice the column contains a newer timestamp, demonstrating that the local data is more up to date     ...as expected.

# Review Highlights
* `metrics.rs` - I am not  100% happy with the readability of the code that generates the HTML and does lookups in the local/global stores. As I move this code to its own module, It will probably make more sense.

### CODE REVIEW

File: src/lib.rs
Comment (ali-bahjati) [inline src/lib.rs:1]: What is the reason for this?

File: Cargo.toml
Comment (ali-bahjati) [inline Cargo.toml:41]: Do you mind commenting why you are using this as a git dependency?

File: src/agent.rs
Comment (ali-bahjati) [inline src/agent.rs:155]: I think it's better to have a metric config and this be the default value for it.

File: src/agent.rs
Comment (ali-bahjati) [inline src/agent.rs:1]: nitpik: I think it's better to move this down.

File: src/agent/metrics.rs
Comment (ali-bahjati) [inline src/agent/metrics.rs:158]: You can use use `humantime::format_duration` to just get the same thing:

```rust
use humantime::format_duration;
let duration_string = format_duration(duration);
```

File: src/agent/metrics.rs
Comment (ali-bahjati) [inline src/agent/metrics.rs:275]: Is there any reason that it's a map?

File: src/agent/metrics.rs
Comment (jayantk) [inline src/agent/metrics.rs:59]: i presume in the future you're going to create another component that aggregates metrics over time (like gets sent info on each update or whatever to track in accumulators), and the data from that component will be accessible here as well?

I think the current information here is a good start to a dashboard, but I'm pretty sure that people will want to see metrics over some time interval. Most problems (like dropped RPC requests, etc) will happen sporadically and you'll want historical data to see whether or not they've been happening.

**Review by jayantk — COMMENTED** [review]
looks good to me. I agree with ali's comments too, so i'll let him approve when you've addressed them.

File: src/agent/metrics.rs
Comment (ali-bahjati) [inline src/agent/metrics.rs:194]: nitpik: maybe in the future I move these to another method to separate the html rendering vs what data to display.

**Review by ali-bahjati — APPROVED** [review]
This looks good. I think it's good to wait a bit for Tom to review this too.
I left some comments and appreciate if you address them before merging.

File: src/agent/metrics.rs
Comment (drozdziak1) [reply src/agent/metrics.rs:275]: BTree* family containers automatically sort their keys as opposed to the regular Hash* implementations. The goal was to get a sorted view every time

File: src/agent/metrics.rs
Comment (drozdziak1) [reply src/agent/metrics.rs:59]: The main piece of Rust boilerplate for adding metrics is that the Rust Prometheus crate does not let you discover existing metrics easily from global metrics state. This state is kept in a `Registry` object, which is the top-level container used to gather and export formatted metrics in bulk. The preferred method for creating/referencing metrics is linking counter/gauge objects to the registry, and using the objects themselves as references to individual metrics, presumably right beside other state variables. 

This pattern was reasonably easy for the attester, where we focused on the overall performance of the system and got away with storing a handful of static metrics as globals.

In the agent, we'll likely need to add the counter objects per-symbol dynamically. I plan to extend global/local store state to hold the new objects and update them with incoming Update* messages. I'ii take good a look at Richard's comments and screenshots to get an idea of what could work well.

File: src/agent/metrics.rs
Comment (drozdziak1) [reply src/agent/metrics.rs:275]: The primary reason I used a map was deduplication. Come to think of it, I should probably detect overwrites on existing entries

File: src/agent/metrics.rs
Comment (drozdziak1) [reply src/agent/metrics.rs:182]: I already broke ground on the metrics front, starting with rearranging the modules. It definitely looks better that way and is coming soon.

File: src/agent/metrics.rs
Comment (drozdziak1) [reply src/agent/metrics.rs]: Done 🫡

File: src/agent.rs
Comment (drozdziak1) [reply src/agent.rs]: Done 🫡

File: src/agent.rs
Comment (drozdziak1) [reply src/agent.rs]: Done 🫡

File: src/lib.rs
Comment (drozdziak1) [reply src/lib.rs:3]: Done 🫡

**Pull request merged**
